Vic2ToHoI4/Source/OutHoi4/OutHoi4World.cpp

```diff
@@ -0,0 +1,549 @@
+	}
+}
+
+
+void HoI4::outputGenericFocusTree(const HoI4FocusTree& genericFocusTree, const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting generic focus tree";
+
+	if (!Utils::TryCreateFolder("output/" + outputName + "/common/national_focus"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/national_focus");
+	}
+
+	outputSharedFocuses(genericFocusTree, "output/" + outputName + "/common/national_focus/shared_focuses.txt");
+}
+
+
+void HoI4::outputCountries(const std::set<Advisor>& activeIdeologicalAdvisors,
+	 const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const allMilitaryMappings& theMilitaryMappings,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting countries";
+
+	if (!Utils::TryCreateFolder("output/" + outputName + "/history"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/history");
+	}
+	if (!Utils::TryCreateFolder("output/" + outputName + "/history/countries"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/history/countries");
+	}
+	if (!Utils::TryCreateFolder("output/" + outputName + "/history/states"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/history/states");
+	}
+	if (!Utils::TryCreateFolder("output/" + outputName + "/history/units"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/history/units");
+	}
+
+	for (const auto& country: countries)
+	{
+		if (country.second->getCapitalState())
+		{
+			const auto& specificMilitaryMappings = theMilitaryMappings.getMilitaryMappings(theConfiguration.getVic2Mods());
+			outputCountry(activeIdeologicalAdvisors, specificMilitaryMappings.getDivisionTemplates(), *country.second);
+		}
+	}
+
+	std::ofstream ideasFile("output/" + outputName + "/interface/converter_ideas.gfx");
+	if (!ideasFile.is_open())
+	{
+		throw std::runtime_error("Could not open output/" + outputName + "/interface/converter_ideas.gfx");
+	}
+
+	ideasFile << "spriteTypes = {\n";
+	for (const auto& country: countries)
+	{
+		if (country.second->getCapitalState())
+		{
+			outputIdeaGraphics(ideasFile, *country.second);
+		}
+	}
+	ideasFile << "\n";
+	ideasFile << "}\n";
+}
+
+
+void HoI4::outputRelations(const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting opinion modifiers";
+
+	if (!Utils::TryCreateFolder("output/" + outputName + "/common/opinion_modifiers"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/opinion_modifiers/");
+	}
+	std::ofstream out("output/" + outputName + "/common/opinion_modifiers/01_opinion_modifiers.txt");
+	if (!out.is_open())
+	{
+		throw std::runtime_error(
+			 "Could not create output/" + outputName + "/common/opinion_modifiers/01_opinion_modifiers.txt");
+	}
+
+	out << "opinion_modifiers = {\n";
+	for (auto i = -200; i <= 200; i++)
```

Comment:
lol, don't tell me this is actually how hoi4 works?

Reply:
Sadly, we know no better way to add a starting opinion modifier. It wants a named modifier instead of a numeric input. So we just define all possible values.

As a bonus we can make the starting modifier decay over time.